### PR TITLE
[rocm-gdb] Add new package

### LINKED ISSUE #237: [rocm-dbgapi] Update to 3.5.0

### LINKED ISSUE #238: [rocm-gdb] Update to 3.5.0

### PR DESCRIPTION
**Pull request by tpkessler**
Resolves #238, resolves #237 

Successfully builds in a clean chroot. We need the "hack" in `rocm-gdb`, otherwise the package does not build. I took it from the PKGBUILD for `gdb`.

### CODE REVIEW

File: rocm-gdb/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    --disable-gdbserver \
+    --disable-sim
+    
+    make
+}
+
+package() {
+    DESTDIR="$pkgdir" make -C "$_dirname/build" install
+    
+    #TODO: License file
```
Comment (acxz) [inline rocm-gdb/PKGBUILD:53]: Can we remove this TODO?

File: rocm-gdb/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+pkgrel=1
+pkgdesc='ROCm source-level debugger for Linux, based on GDB'
+arch=('x86_64')
+url='https://github.com/ROCm-Developer-Tools/ROCgdb'
+license=('custom: GPL with AMD copyright')
+depends=('rocm-dbgapi' 'python' 'guile2.0' 'ncurses' 'expat' 'xz' 'zlib' 'mpfr' 'source-highlight' 'babeltrace')
+makedepends=('texinfo')
+source=("$pkgname-$pkgver.tar.gz::$url/archive/rocm-$pkgver.tar.gz")
+sha256sums=('cf36d956e84c7a5711b71f281a44b0a9708e13e941d8fca0247d01567e7ee7d1')
+_dirname=$(basename "$url")-$(basename "${source[0]}" ".tar.gz")
```
Comment (ranisalt) [inline rocm-gdb/PKGBUILD:13]: Quotes

File: rocm-gdb/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    --with-guile=guile-2.0 \
+    --with-expat \
+    --with-system-zlib \
+    --with-babeltrace \
+    --with-lzma \
+    --disable-gdbtk \
+    --disable-ld \
+    --disable-gas \
+    --disable-gdbserver \
+    --disable-sim
```
Comment (ranisalt) [inline rocm-gdb/PKGBUILD:45]: One level of indentation here would make it much better to read.

File: rocm-gdb/PKGBUILD (hunk 1 above)
Comment (tpkessler) [reply rocm-gdb/PKGBUILD:53]: Sure.

File: rocm-gdb/PKGBUILD (hunk 2 above)
Comment (tpkessler) [reply rocm-gdb/PKGBUILD:13]: Done

File: rocm-gdb/PKGBUILD (hunk 3 above)
Comment (tpkessler) [reply rocm-gdb/PKGBUILD:45]: You're right. I've added one more level of indentation.

File: rocm-gdb/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        --disable-gas \
+        --disable-gdbserver \
+        --disable-sim
+    
+    make
+}
+
+package() {
+    DESTDIR="$pkgdir" make -C "$_dirname/build" install
+}
```
Comment (ranisalt) [inline rocm-gdb/PKGBUILD:52]: I think when acxz said to remove TODO, he meant adding the license in place.

Install `COPYING` same way we do with the other packages.

Comment (tpkessler) [reply rocm-gdb/PKGBUILD:52]: I was not sure what license I should put in the package. From the upstream repo:
>Update copyright and license
>
>Add GPL license with an AMD copyright in rocm gdb tests related files.

So is it the usual GPL (tests are not included in the final package) ?

Comment (acxz) [reply rocm-gdb/PKGBUILD:52]: Yeah the license is GPL but the copyright is still AMD, I believe this is fine.

**Pull request merged**